kinto/plugins/openid/__init__.py

```diff
@@ -0,0 +1,193 @@
+import re
+
+import requests
+from jose import jwt
+from pyramid import authentication as base_auth
+from pyramid.interfaces import IAuthenticationPolicy
+from zope.interface import implementer
+
+from kinto.core import logger
+from kinto.core import utils as core_utils
+from kinto.core.openapi import OpenAPI
+
+from .utils import fetch_openid_config
+
+
+@implementer(IAuthenticationPolicy)
+class OpenIDConnectPolicy(base_auth.CallbackAuthenticationPolicy):
+    def __init__(self, issuer, client_id, realm='Realm', **kwargs):
+        self.realm = realm
+        self.issuer = issuer
+        self.client_id = client_id
+        self.client_secret = kwargs.get('client_secret', '')
+        self.header_type = kwargs.get('header_type', 'Bearer')
+        self.userid_field = kwargs.get('userid_field', 'sub')
+        self.verification_ttl = int(kwargs.get('verification_ttl_seconds', 86400))
+
+        # Fetch OpenID config (at instantiation, ie. startup)
+        self.oid_config = fetch_openid_config(issuer)
+
+        self._jwt_keys = None
+
+    def unauthenticated_userid(self, request):
+        """Return the userid or ``None`` if token could not be verified.
+        """
+        settings = request.registry.settings
+        hmac_secret = settings['userid_hmac_secret']
+
+        authorization = request.headers.get('Authorization', '')
+        try:
+            authmeth, payload = authorization.split(' ', 1)
+        except ValueError:
+            return None
+
+        if authmeth.lower() != self.header_type.lower():
+            return None
+
+        try:
+            # Bearer+OIDC id_token=jwt, access_token=bearer
```

Comment:
Is this on purpose? I though we said that we would only send Bearer token and not id_tokens anymore?